ibis/common/grounds.py

```diff
@@ -201,7 +214,7 @@ def __cached_equals__(self, other):
             return True
 
         # type comparison should be cheap
-        if type(self) != type(other):
+        if type(self) is not type(other):
```

Comment:
Is this change meaningful in some way? When is `!=` not equivalent to `is not` in this context? Dynamically generated types?

Reply:
It's just more expressive syntactically.